python/llm/src/ipex_llm/transformers/models/stablelm.py

```diff
@@ -253,3 +316,159 @@ def stablelm_attention_forward(
+                                                                 sin,
+                                                                 cos,
+                                                                 "stablelm",
+                                                                 position_ids)
+    else:
+        query_rot, key_rot = apply_rotary_pos_emb(query_rot,
+                                                  key_rot,
+                                                  cos,
+                                                  sin,
+                                                  position_ids,
+                                                  "stablelm")
+
+    # [batch_size, seq_length, num_heads, head_dim]
+    query_states = torch.cat((query_rot, query_pass), dim=-1)
+    key_states = torch.cat((key_rot, key_pass), dim=-1)
+
+    kv_seq_len = key_states.shape[-2]
+    if len(past_key_value.key_cache) <= self.layer_idx:
+        attn_weights = torch.matmul(query_states, key_states.transpose(2, 3)) / math.sqrt(self.head_dim)
+
+        if attn_weights.size() != (bsz, self.num_heads, q_len, kv_seq_len):
+            raise ValueError(
+                f"Attention weights should be of size {(bsz, self.num_heads, q_len, kv_seq_len)}, but is"
+                f" {attn_weights.size()}"
+            )
+
+        if attention_mask is not None:
+            if attention_mask.size() != (bsz, 1, q_len, kv_seq_len):
+                raise ValueError(
+                    f"Attention mask should be of size {(bsz, 1, q_len, kv_seq_len)}, but is {attention_mask.size()}"
+                )
+            attn_weights = attn_weights + attention_mask
+
+        # upcast attention to fp32
+        attn_weights = nn.functional.softmax(attn_weights, dtype=torch.float32, dim=-1).to(query_states.dtype)
+        attn_weights = self.attention_dropout(attn_weights)
+
+        attn_output = torch.matmul(attn_weights, value_states)
+
+        if attn_output.size() != (bsz, self.num_heads, q_len, self.head_dim):
+            raise ValueError(
+                f"`attn_output` should be of size {(bsz, self.num_heads, q_len, self.head_dim)}, but is"
+                f" {attn_output.size()}"
+            )
+        if use_cache:
+            cache_kwargs = None
+            key_states, value_states = past_key_value.update(key_states, value_states,
+                                                             self.layer_idx, cache_kwargs,
+                                                             new_layout=True)
+    else:
+        cache_kwargs = None  # Specific to RoPE models
+        key_states, value_states = past_key_value.update(key_states, value_states,
+                                                         self.layer_idx, cache_kwargs,
+                                                         new_layout=True)
+        kv_seq_len = key_states.shape[-2]
+        if query_states.size(2) != 1 or query_states.device.type != 'xpu' or \
```

Comment:
The output will be all `!!!!!!!!`, as the second hidden_states values are all `nan`. It's caused by sdp_fp8.
If we add `or self.layer_idx < 7`  here, the generation output will right.

Reply:
self.layer_idx is the number of StableLmAttention. For stablelm-2-zephyr-1_6b, total is 24. 
Model stucture:
```
StableLmForCausalLM(
  (model): StableLmModel(
    (embed_tokens): LLMEmbedding(100352, 2048)
    (layers): ModuleList(
      (0-23): 24 x StableLmDecoderLayer(
        (self_attn): StableLmAttention(
          (o_proj): LowBitLinear(in_features=2048, out_features=2048, bias=False)
          (attention_dropout): Dropout(p=0.0, inplace=False)
          (rotary_emb): StableLmRotaryEmbedding()
          (qkv_proj): LowBitLinear(in_features=2048, out_features=6144, bias=True)
        )
        (mlp): StableLmMLP(
          (gate_proj): LowBitLinear(in_features=2048, out_features=5632, bias=False)
          (up_proj): LowBitLinear(in_features=2048, out_features=5632, bias=False)
          (down_proj): LowBitLinear(in_features=5632, out_features=2048, bias=False)
          (act_fn): SiLU()
        )
        (input_layernorm): LayerNorm((2048,), eps=1e-05, elementwise_affine=True)
        (post_attention_layernorm): LayerNorm((2048,), eps=1e-05, elementwise_affine=True)
        (dropout): Dropout(p=0.0, inplace=False)
      )
    )
    (norm): LayerNorm((2048,), eps=1e-05, elementwise_affine=True)
  )
  (lm_head): LowBitLinear(in_features=2048, out_features=100352, bias=False)
)

```